Backport missing PRs from Bitcoin 0.13 - Part 1

codablock opened the pull request:
This is the first batch of backported PRs from Bitcoin which are still missing in Dash.

I've got a total of ~780 PRs (including 0.13+0.14 PRs) in my local branches, so the amount of changed code is huge. I was thinking alot about how large the individual batches should be and decided to make them ~50 PRs per batch. This way we'll have around 16 huge PRs instead of hundreds of small ones. I understand that reviewing these batches will be hell, but having smaller batches won't really change the amount of work required and only prolong everything.

I've gone through all PRs and tried my best to omit PRs for features we do not want in Dash or which conflict with Dash specific features/changes. If I mistakenly included something that should have been omitted, please tell me and I remove these changes.

As the splitting of main.cpp->net_processing.cpp/validation.cpp was done out-of-order, I had to do a lot of manual copy+paste conflict resolution. This poses a high risk of loosing changes. My plan is to do a full-tree diff when all PRs are merged and if anything was lost, I'll hopefully catch it this way.

codablock commented:
I disabled check-doc.py and Mac builds on Travis for now. I'll fix errors reported by check-doc.py after all PRs have been merged.

UdjinM6 commented on configure.ac:
```diff
@@ -6,6 +6,8 @@ define(_CLIENT_VERSION_REVISION, 2)
 define(_COPYRIGHT_YEAR, 2017)
+define(_COPYRIGHT_HOLDERS,[The %s developers])
+define(_COPYRIGHT_HOLDERS_SUBSTITUTION,[Dash Core])
 AC_INIT([Dash Core],[_CLIENT_VERSION_MAJOR._CLIENT_VERSION_MINOR._CLIENT_VERSION_REVISION],[https://github.com/dashpay/dash/issues],[dashcore])
```
Needs `[https://dash.org/]` like in original PR https://github.com/bitcoin/bitcoin/pull/7192/files#diff-67e997bcfdac55191033d57a16d1408aR11

codablock replied:
Fixed and squashed

UdjinM6 commented on contrib/macdeploy/custom_dsstore.py:
```diff
@@ -0,0 +1,60 @@
+alias.volume.name = package_name_ns
+alias.volume.posix_path = '/Volumes/' + package_name_ns
+alias.volume.disk_image_alias.target.filename = package_name_ns + '.temp.dmg'
+alias.volume.disk_image_alias.target.carbon_path = 'Macintosh HD:Users:\x00bitcoinuser:\x00Documents:\x00bitcoin:\x00bitcoin:\x00' + package_name_ns + '.temp.dmg'
```
Lines 48, 49, 57 should probably be dashified

codablock replied:
Fixed and squashed

UdjinM6 commented on share/qt/Info.plist.in:
```diff
@@ -17,7 +17,7 @@
 
   <key>CFBundleGetInfoString</key>
-  <string>@CLIENT_VERSION_MAJOR@.@CLIENT_VERSION_MINOR@.@CLIENT_VERSION_REVISION@, Copyright © 2009-@COPYRIGHT_YEAR@ The Bitcoin Core developers, 2014-@COPYRIGHT_YEAR@ The Dash Core developers</string>
+  <string>@CLIENT_VERSION_MAJOR@.@CLIENT_VERSION_MINOR@.@CLIENT_VERSION_REVISION@, Copyright © 2009-@COPYRIGHT_YEAR@ @COPYRIGHT_HOLDERS_FINAL@</string>
```
Should keep Bitcoin Core copyright here

codablock replied:
Fixed and squashed

UdjinM6 commented on src/clientversion.h:
```diff
@@ -38,7 +38,7 @@
 
 //! Copyright string used in Windows .rc files
-#define COPYRIGHT_STR "2009-" STRINGIZE(COPYRIGHT_YEAR) " The Bitcoin Core Developers, 2014-" STRINGIZE(COPYRIGHT_YEAR) " The Dash Core Developers"
+#define COPYRIGHT_STR "2009-" STRINGIZE(COPYRIGHT_YEAR) " " COPYRIGHT_HOLDERS_FINAL
```
Should keep Bitcoin Core copyright here

codablock replied:
Fixed and squashed

UdjinM6 commented on src/qt/res/dash-qt-res.rc:
```diff
@@ -19,13 +19,13 @@ BEGIN
         BEGIN
             VALUE "CompanyName",        "Dash"
-            VALUE "FileDescription",    "Dash Core (GUI node for Dash)"
+            VALUE "FileDescription",    PACKAGE_NAME " (GUI node for Bitcoin)"
```
s/Bitcoin/Dash/

codablock replied:
Fixed and squashed

codablock commented:
Thanks for the review! :+1: 

Regarding your 3 general comments:
1. Agree, it's a fix that should normally not be here. I stumbled across this one while fixing Travis issues with this PR, so I decided to go the short path and just fix it as part of this PR. I have a personal policy in regard to such changes: If I stumble across minor things which have no or only minor effects (like typos, or like this one that has clearly no functional effect), I fix them in-line while working on the branch that I'm currently working on. I prefer this because I've seen too many developers who tend to ignore minor things ("it's just a minor thing...no one cares, right? Let someone else fix it later...") rather then going the officially correct way of creating separate PRs for such things.
2. Added missing change and squashed into the commit. I hope such things didn't slip through that often...my workflow was a little bit undefined when I started and got better and better after each PR.
3. Rechecked where it came from. The commit/PR description comes from #7482 but I had merge conflicts while backporting this one. I realized that the PR got obsolete due to #7804 being already backported to Dash and discarded the PR. Looks like somehow the commit message got not discarded while I continued on the next PR, which was #7509. I fixed this by re-backporting the PR and putting it into the same position.

Inline-comments have been fixed and squashed.

UdjinM6 commented:
Looks better :)

Found few more issues issues though:
![untitled](https://user-images.githubusercontent.com/1935069/33835921-fcc45fe2-de98-11e7-82be-ce78db8a29e3.png)

Proposing 2 fixes:
- make `CopyrightHolders` more generic https://github.com/UdjinM6/dash/commit/f9b2a775cad5a7252b109f844a62687bc44433c5
- fix splashscreen (shrink `titleCopyrightVSpace`) https://github.com/UdjinM6/dash/commit/7a120d4445de9fdbad7a1475a86c1e961e3bd34f

With these fixes applied:
![untitled2](https://user-images.githubusercontent.com/1935069/33835763-905fa370-de98-11e7-8862-c9fe1ee92d0e.png)

PS. Pls add fixes in a separate commit on top of HEAD instead of rebasing&squashing.

codablock commented:
Thanks for the 2 fixed you provided, cherry-picked them on top :)

UdjinM6 commented:
ACK, but pls don't merge this till 12.2.2 is tagged/released

spencerlievens commented:
I also tested this branch and the listed Masternodes became NEW_START_ENABLED, all. I assume that if all nodes updated to this same code, then they would in fact report as ENABLED. This needs looking at.
